### PR TITLE
Feature/section

### PR DESCRIPTION
**Pull request by rinkishivani**

### CODE REVIEW

File: products/views.py [hunk 1]
```diff
@@ -550,3 +561,110 @@ def delete_banner(request, pk):
...
+
+    sections = Section.objects.all().order_by('order')
+    search = request.GET.get('search')
+    if search is None:
+        search = ""
+    if search:
+        sections = Section.objects.filter(Q(name__icontains=search) | Q(order__icontains=search)
+                                          | Q(is_active__icontains=search))
+    else:
+        sections = sections
```
Comment (palakparmartntra) [inline products/views.py]: remove serach condition, and remove `if search is None:`

File: products/views.py [hunk 2]
```diff
@@ -550,3 +561,110 @@ def delete_banner(request, pk):
...
+
+    page = Paginator(sections, 3)
+    page_number = request.GET.get('page')
+    try:
+        page_obj = page.get_page(page_number)
+    except PageNotAnInteger:
+        page_obj = page.page(1)
+    except EmptyPage:
+        page_obj = page.page(page.num_pages)
```
Comment (palakparmartntra) [inline products/views.py:582]: create common pagination class.

File: products/views.py [hunk 3]
```diff
@@ -550,3 +561,110 @@ def delete_banner(request, pk):
...
+    }
+    return render(request, 'section/view_section.html', context)
+
+
+@login_required
+def update_section(request, pk):
+    """ To update section details """
+
+    if not request.user.is_superuser:
+        raise Http404
```
Comment (palakparmartntra) [inline products/views.py]: move this to the common class or decorator.

File: products/messages.py [hunk 4]
```diff
@@ -15,6 +15,7 @@ class SectionFormErrorMessages:
     SECTION_NAME = "Section name is required"
     ORDER = "Section priority is required"
     ORDER_NOT_UNIQUE = "Priority of the section should be unique!"
+    INVALID_FILE_TYPE = "Upload an excel file"
```
Comment (palakparmartntra) [inline products/messages.py]: give proper message for invalid file type. @rinkishivani @Yshah112

File: products/views.py (hunk 3 above)
Comment (rinkishivani) [reply products/views.py]: Done

File: products/views.py (hunk 1 above)
Comment (rinkishivani) [reply products/views.py]: Done

**Review by palakparmartntra — APPROVED** [review]

**Pull request merged**